packages/mask/src/components/shared/WalletStatusBox/TransactionList.tsx

```diff
@@ -0,0 +1,197 @@
+import { noop } from 'lodash-unified'
+import { FC, forwardRef, useCallback, useEffect, useMemo, useState } from 'react'
+import { useAsync } from 'react-use'
+import Services from '../../../extension/service'
+import type { RecentTransaction } from '../../../plugins/Wallet/services/transaction'
+import { useI18N } from '../../../utils'
+
+const useStyles = makeStyles()((theme) => ({
+    list: {
+        borderRadius: 8,
+        overflow: 'hidden',
+        backgroundColor: theme.palette.background.paper,
+        padding: 0,
+    },
+    listItem: {
+        height: 54,
+        backgroundColor: theme.palette.background.paper,
+        '&:nth-child(even)': {
+            backgroundColor: theme.palette.background.default,
+        },
+    },
+    transaction: {
+        width: '100%',
+    },
+    methodName: {
+        fontWeight: 'bold',
+        fontSize: 14,
+        textTransform: 'capitalize',
+        textOverflow: 'ellipsis',
+        overflow: 'hidden',
+        whiteSpace: 'nowrap',
+    },
+    cell: {
+        fontSize: 14,
+        display: 'flex',
+        alignItems: 'center',
+        padding: theme.spacing(0, 0.5),
+        color: theme.palette.text.primary,
+        boxSizing: 'border-box',
+    },
+    link: {
+        fill: 'none',
+    },
+    linkIcon: {
+        fill: 'none',
+        width: 12,
+        height: 12,
+        marginLeft: theme.spacing(0.5),
+    },
+}))
+
+const statusTextColorMap: Record<TransactionStatusType, string> = {
+    [TransactionStatusType.NOT_DEPEND]: '#FFB915',
+    [TransactionStatusType.SUCCEED]: '#60DFAB',
+    [TransactionStatusType.FAILED]: '#FF5F5F',
+    [TransactionStatusType.CANCELLED]: '#FF5F5F',
+}
+const getContractFunctionName = async (data: string | undefined) => {
+    if (!data) return null
+    const sig = data.slice(0, 10)
+    const name = await Services.Ethereum.getContractFunctionName(sig)
+    return name ? name.replace(/_/g, ' ') : 'Contract Interaction'
+}
+
+interface TransactionProps extends GridProps {
+    chainId: ChainId
+    transaction: RecentTransaction
+    onClear?(tx: RecentTransaction): void
+}
+const Transaction: FC<TransactionProps> = ({ chainId, transaction: tx, onClear = noop, ...rest }) => {
+    const { t } = useI18N()
+    const { Utils } = useWeb3State()
+    const { classes, theme } = useStyles()
+
+    const statusTextMap: Record<TransactionStatusType, string> = {
+        [TransactionStatusType.NOT_DEPEND]: t('recent_transaction_pending'),
+        [TransactionStatusType.SUCCEED]: t('recent_transaction_success'),
+        [TransactionStatusType.FAILED]: t('recent_transaction_failed'),
+        [TransactionStatusType.CANCELLED]: t('recent_transaction_cancelled'),
+    }
+
+    const web3 = useWeb3()
+    const { value: targetAddress } = useAsync(async () => {
+        if (tx.receipt?.contractAddress) return tx.receipt.contractAddress
+        if (tx.payload?.params?.[0].to) return tx.payload.params[0].to as string
```

Comment:
You can use the lodash head function

Reply:
Access `tx.payload.params` twice, don't wanna use head twice.